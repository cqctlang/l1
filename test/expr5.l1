mkadd = @lambda(n){
      return @lambda(m){ return m+n; };
};

callit = @lambda(fn,m){
       return fn(m);
};

add10 = mkadd(10);
print(callit(add10,4));
print(callit(add10,8));

