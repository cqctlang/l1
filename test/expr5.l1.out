entry: 0 params, 0 locals, 0 temps
L3: 1 params, 0 locals, 0 temps
L6: 1 params, 0 locals, 0 temps
L10: 2 params, 0 locals, 0 temps
<cval 14>
<cval 18>
